docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,166 @@
+- We would be removing the fixed labels that can be propagated from the clusterDeployment, i.e.`cluster_type`, `platform` and `version`. This means, if admins want these labels for the metrics, they will have to define them via  metricsConfig.metricsToReport[].additionalClusterDeploymentLabels. This would only be a breaking change if the admins suddenly cut over from deprecated/old way of metricsConfig to the new metricsToReport, sufficient documentation should help though.
+- We would be renaming `hive_cluster_deployment_install_failure_total` and `hive_cluster_deployment_install_success_total` to ensure it ends with `seconds`, as it is a duration-based metric. Note that this will result in cloning the existing metric for the deprecation period, because if `metricsToReport` isn't present in the hiveConfig, then the _old_ way has to still work.
+- We would no longer panic when the metricsConfig fails validation, instead we would be updating hiveConfig.Status.Conditions[ReadyCondition] to false, with appropriate reason and message. See [Failure Modes](#failure-modes) for more details.
+
+#### Deprecation
+
+In order to implement this change, we will have to deprecate the existing hiveConfig.spec.metricsConfig.metricsWithDuration and hiveConfig.spec.metricsConfig.additionalClusterDeploymentLabels.
+As a way to handle the deprecation, we will assert that in a hiveConfig, you can either provide the new configurations (aka `metricsToReport`) or the old way of hiveConfig.spec.metricsConfig.metricsWithDuration and hiveConfig.spec.metricsConfig.additionalClusterDeploymentLabels. If there's an empty metricsConfig, we would default to _old_ behaviour for the period of deprecation.
+Note that only the old design will support the mapping of camelCase key to metric for minimumDuration, the new way will require full metric names only. The old design will log `hive_cluster_deployment_install_failure_total` and `hive_cluster_deployment_install_success_total`, in the new design, they will be renamed to `hive_cluster_deployment_install_failure_total_seconds` and `hive_cluster_deployment_install_success_total_seconds`. 
+
+#### Failure modes
+
+These situations will result in hive-operator failing to deploy the controllers. It will set hiveConfig.Status.Conditions[ReadyCondition] to `"False"` with an appropriate error message.
+-  `metricsToReport[].metricNames[$name]` doesn't exist as a metric.
+- `metricNames` lists a metric without duration but `minimumDuration` is specified in the same `metricsToReport` entry.
+- the same metric is mentioned more than once across all `metricsToReport[].metricNames[]`.
+- additionalClusterDeploymentLabels key conflict with an existing fixed label key for that metric.
+- if deprecated methods are used along with the new metricsToReport. You can either choose the _old_ way or the _new_ way.
+
+
+### Risks and Mitigations
```

Comment:
Do you have processes or tools in mind, to prevent acidentialy enabling a lot of metrics?
Or asked in another way, can you predict the impact/outcome of a config change?